rfc-parser/src/main/java/tech/pantheon/yanginator/parser/services/GrammarKitRFCUtils.java

```diff
@@ -0,0 +1,342 @@
+     * These commentaries are replaced by actual rules.
+     *
+     * @param lines the list of strings in which string rules are to be replaced.
+     * @return the list of strings with string rules replaced.
+     */
+    public static List<String> rewriteStringRules(final List<String> lines) {
+        final Pattern rulePattern = Pattern.compile(".*(?<statementStart>(<[\\s+\\w].*))$");
+        List<String> result = new ArrayList<>();
+        for (int i = 0; i < lines.size(); i++) {
+            String firstLineOfRule = lines.get(i);
+            Matcher matcher = rulePattern.matcher(firstLineOfRule.trim());
+            if (matcher.matches()) {
+                String originalString = matcher.group("statementStart");
+                String quotes = "\"\\\"\"";
+                String secondLineOfRule = lines.get(i + 1);
+                if (secondLineOfRule.contains("-arg")) {
+                    secondLineOfRule = secondLineOfRule.trim().replace(">", quotes);
+                    String replacement = "";
+                    firstLineOfRule = firstLineOfRule.replace(originalString, replacement.concat(quotes).concat(" " + secondLineOfRule));
+                } else {
+                    firstLineOfRule = firstLineOfRule
+                            .replace("<", "//")
+                            .concat(" " + secondLineOfRule.trim())
+                            .replace(">", "");
+                }
+                i++;
+            }
+            result.add(firstLineOfRule);
+        }
+
+        return result;
+    }
+
+    /**
+     * Replaces each occurrence of a hexadecimal range of values with corresponding characters.
+     *
+     * @param lines the list of strings in which range of hexadecimal values will be replaced.
+     * @return the list of strings in which range of hexadecimal values are replaced by characters.
+     */
+    public static List<String> replaceHexadecimalRange(List<String> lines) {
+        List<String> result = new ArrayList<>();
+        final Pattern hexPatternRange = Pattern.compile("(?<hexadecimalRange>(%x\\d{1,2}\\w-\\d{1,2}\\w))");
+
+        for (String line : lines) {
+            Matcher matcher = hexPatternRange.matcher(line.trim());
+            while (matcher.find()) {
+                String originalHexDef = matcher.group("hexadecimalRange").trim();
+                String originalHexDefWithZero = originalHexDef.trim().replaceAll("%", "0");
+                String hexLowerBoundaryString = originalHexDefWithZero.substring(0, 4);
+                int hexLowerBoundary = Integer.decode(hexLowerBoundaryString);
+                int hexUpperBoundary = Integer.parseInt(originalHexDefWithZero.substring(5).toLowerCase(), 16);
+                int hex = hexLowerBoundary;
+                StringBuilder replacement = new StringBuilder();
+                while (hex <= hexUpperBoundary) {
+                    if (hex == 34 | hex == 92) {
```

Comment:
34 -> '"'
92 -> '\\'
to make it clear what is going on
also "||" not "|"

Reply:
Would you be please more specific, how should I change this? Should I add comment to clarify what is going on, or just change the following line, so " and \ are enclosed in single quotes?

Reply:
the single-quote thing primarily.
the other is use of bit-wise | operator vs. logical || operator -- use the latter here (and everywhere else)
the reason is that | always evaluates each part, whereas || stops as soon as it arrives at 'true' -- just as & vs &&, where && stops as soon as it arrives at 'false'

Reply:
Fixed.